packages/wdio-allure-reporter/src/index.js

```diff
@@ -101,12 +101,23 @@ class AllureReporter extends WDIOReporter {
         const currentTest = this.allure.getCurrentTest()
 
         if (!this.isMultiremote) {
-            const { browserName, deviceName } = this.capabilities
-            const targetName = browserName || deviceName || cid
-            const browserstackVersion = this.capabilities.os_version || this.capabilities.osVersion
-            const version = browserstackVersion || this.capabilities.version || this.capabilities.platformVersion || ''
-            const paramName = deviceName ? 'device' : 'browser'
-            const paramValue = version ? `${targetName}-${version}` : targetName
+            const { browserName, device, deviceName, version, platformVersion } = this.capabilities
+
+            let targetName = browserName || device || deviceName || cid
+            let paramVersion = version || platformVersion || ''
+            let paramName = device || deviceName ? 'device' : 'browser'
+
+            if (this.config.hostname.includes('browserstack')) {
+                // eslint-disable-next-line camelcase
+                const { browser_name: browserNameLegacy, os_version: osVersionLegacy, osVersion: osVersionW3C } = this.capabilities
+                const { version: mobileVersion } = this.capabilities.mobile
```

Comment:
Instead of destructuring twice, you can do something like this
```js
const {
  browser_name: browserNameLegacy,
  os_version: osVersionLegacy,
  osVersion: osVersionW3C,
  mobile: { version: mobileVersion }
} = this.capabilities
```